src/View/Helper/FormHelper.php

```diff
@@ -1649,7 +1649,7 @@ public function postLink($title, $url = null, array $options = [])
 
         $formName = str_replace('.', '', uniqid('post_', true));
         $formOptions = [
-            'action' => $this->Url->build($url),
+            'action' => Router::url($url),
```

Comment:
It is likely that the lastAction property is the problem. If it also has the html encoded version, there could be issues.